core/templates/dev/head/collection_editor/Services.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Standalone services for the exploration editor page.
```

Comment:
I agree. This is quite repetitive. My initial idea is you should avoid using a custom data handler for the editor and, instead, use the viewer's editor to grab the collection data. You will still need a data handler later on, but only for saving. Although this sounds nice, the problem that might come up is the editor might need additional information that the collection viewer does not (see editor.ExplorationHandler._get_exploration_data versus reader.ExplorationHandler.get). I'm not convinced whether we should have a second get handler for the additional information the collection editor might need not provided by the viewer's data handler.

@seanlip do you have any thoughts on this? I lean toward having 2 different get handlers to minimize duplication, but you might have a better idea on solving this problem. Thanks!